src/tests/JIT/HardwareIntrinsics/General/Shared/VectorDotTest.template

```diff
@@ -292,12 +292,20 @@ namespace JIT.HardwareIntrinsics.General
             bool succeeded = true;
 
             {RetBaseType} actualResult = default;
+            {RetBaseType} intermResult = default;
 
             for (var i = 0; i < Op1ElementCount; i++)
             {
-                actualResult += ({RetBaseType})(left[i] * right[i]);
+                if ((i % Vector128<{Op1BaseType}>.Count) == 0)
+                {
+                    actualResult += intermResult;
+                    intermResult = default;
```

Comment:
Is this to "simulate" what `Dot` implementation does with floating point element type vectors?
Say, for `Vector256<double>` the value is computed as
 `((left[0] * right[0]) + (left[1] * right[1])) + ((left[2] * right[2]) + (left[3] * right[3]))`

Reply:
Right. For `Vector256<T>` it independently does sum on the lower/upper halves and then those halves are added together.

This means that the vectorized `Dot` would potentially differ from the software `Dot`. So I made sure they stayed consistent.